### PR TITLE
(wip): add basic go example

### PR DESCRIPTION
**Pull request by notrab**

### CODE REVIEW

File: with-go/main.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	req := graphql.NewRequest(`
+    { 
+			products {
+				slug
+				name
+			}
+		}
+	`)
+
+	ctx := context.Background()
```
Comment (BrunoScheufler) [inline with-go/main.go:29]: You could use r.Context here as this handles cases where the request gets aborted

File: with-go/main.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	"html/template"
+	"log"
+	"net/http"
+
+	"github.com/machinebox/graphql"
+)
+
+func main() {
+	http.HandleFunc("/", getProducts)
+	http.ListenAndServe(":8080", nil)
```
Comment (BrunoScheufler) [inline with-go/main.go:14]: This function returns an error if the server could not be started up, you could check it with an if statement and conditionally log or panic to make sure the user knows when something's going wrong!

```go
err := http.ListenAndServe(":8080", nil)
if err != nil {
  panic(err)
}
```

File: with-go/main.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+func getProducts(w http.ResponseWriter, r *http.Request) {
+	client := graphql.NewClient("https://api-eu-central-1.graphcms.com/v2/ck8sn5tnf01gc01z89dbc7s0o/master")
```
Comment (BrunoScheufler) [inline with-go/main.go:18]: Potentially you could move the client creation into the main function, but that's not required

File: with-go/main.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+	}
+
+	type ProductPageData struct {
+		Products []Product
+	}
+
+	var respData ProductPageData
+
+	if err := client.Run(ctx, req, &respData); err != nil {
+		log.Fatal(err)
```
Comment (BrunoScheufler) [inline with-go/main.go:43]: We'd need to send an error response to the client here, maybe another template?

File: with-go/main.go (hunk 3 above)
Comment (notrab) [reply with-go/main.go:18]: This would be useful actually. I want to introduce another route for /products/:slug

File: with-go/main.go [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+	http.ListenAndServe(":8080", nil)
+}
+
+func getProducts(w http.ResponseWriter, r *http.Request) {
+	client := graphql.NewClient("https://api-eu-central-1.graphcms.com/v2/ck8sn5tnf01gc01z89dbc7s0o/master")
+
+	req := graphql.NewRequest(`
+    { 
+			products {
+				slug
```
Comment (pasqualino) [inline with-go/main.go:23]: The `slug` is not being used in the template. Should we add it?

File: with-go/main.go [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+				name
+			}
+		}
+	`)
+
+	ctx := context.Background()
+
+	type Product struct {
```
Comment (pasqualino) [inline with-go/main.go:31]: Maybe we could move these type definitions outside this function, so it's easier to read.

File: with-go/main.go (hunk 5 above)
Comment (notrab) [reply with-go/main.go:23]: @pasqualino I'll be using `slug` as part of the `index.html` to show links to each product page, so it's fine to stay til I get to that.

File: with-go/main.go (hunk 6 above)
Comment (notrab) [reply with-go/main.go:31]: Agreed

**Pull request closed without merging**